homeassistant/helpers/translation.py

```diff
@@ -162,32 +89,26 @@ async def _async_get_component_strings(
     """Load translations."""
     translations_by_language: dict[str, dict[str, Any]] = {}
     # Determine paths of missing components/platforms
-    files_to_load_by_language: dict[str, dict[str, str]] = {}
+    files_to_load_by_language: dict[str, dict[str, pathlib.Path]] = {}
     loaded_translations_by_language: dict[str, dict[str, Any]] = {}
     has_files_to_load = False
     for language in languages:
-        files_to_load: dict[str, str] = {}
+        files_to_load: dict[str, pathlib.Path] = {}
         files_to_load_by_language[language] = files_to_load
         translations_by_language[language] = {}
 
         for comp in components:
-            domain, _, platform = comp.partition(".")
+            domain = comp.partition(".")[0]
             if (
                 not (integration := integrations.get(domain))
                 or not integration.has_translations
             ):
                 continue
 
-            if platform and integration.is_built_in:
-                # Legacy state translations are no longer used for built-in integrations
-                # and we avoid trying to load them. This is a temporary measure to allow
-                # them to keep working for custom integrations until we can fully remove
-                # them.
-                continue
-
-            if path := component_translation_path(comp, language, integration):
```

Comment:
We no longer need to call `component_translation_path` because `has_translations` already tells us if the integration has translations or not and since it now is reduce down to a single path, I removed the function

Reply:
Well that broke all the tests so lets do that in a follow up